hummingbot/cli/utils/exchange_rate_conversion.py

```diff
@@ -28,84 +33,114 @@ def logger(cls) -> logging.Logger:
-            self._exchange_rate = {k: float(v.get("default", 1.0)) for k, v in self._exchange_rate_config.items()}
-            self._exchange_rate_fetcher_config = {e[0]: {"default": None, "source": e[1]}
-                                                  for e in exchange_rate_fetcher_config}
-            self._exchange_rate_manual = {k: None for k, v in self._exchange_rate_fetcher_config.items()} 
+                conversion_required = self._exchange_rate_config_override.get("conversion_required", {})
+                global_config = self._exchange_rate_config_override.get("global_config", {})
+
+            self._exchange_rate_config = {
+                "conversion_required": conversion_required,
+                "global_config": {**global_config, **conversion_required}
+            }
+            self._exchange_rate = {k: v["default"] for k, v in self._exchange_rate_config["global_config"].items()}
         except Exception:
             self.logger().error("Error initiating config for exchange rate conversion.", exc_info=True)
 
     def adjust_token_rate(self, symbol: str, price: float) -> float:
+        """
+        Returns the USD rate of a given token if it is found in conversion_required config
+        :param symbol:
+        :param price:
+        :return:
+        """
         if not self._started:
             self.start()
-
-        if self._exchange_rate.get(symbol, None) is not None:
+        if symbol in self._exchange_rate_config["conversion_required"] and symbol in self._exchange_rate:
             return self._exchange_rate[symbol] * price
         else:
             return price
 
     def convert_token_value(self, amount: float, from_currency: str, to_currency: str):
+        """
+        Converts a token amount to the amount of another token with equivalent worth
+        :param amount:
+        :param from_currency:
+        :param to_currency:
+        :return:
+        """
         if not self._started:
             self.start()
         # assume WETH and ETH are equal value
         if from_currency == "ETH" and to_currency == "WETH" or from_currency == "WETH" and to_currency == "ETH":
             return amount
-        from_currency_usd_rate = self._exchange_rate_manual.get(from_currency, None)
-        to_currency_usd_rate = self._exchange_rate_manual.get(to_currency, None)
-        if from_currency_usd_rate is None or to_currency_usd_rate is None:
+        from_currency_usd_rate = self._exchange_rate.get(from_currency, NaN)
+        to_currency_usd_rate = self._exchange_rate.get(to_currency, NaN)
+        if math.isnan(from_currency_usd_rate) or math.isnan(to_currency_usd_rate):
             raise ValueError(f"Unable to convert '{from_currency}' to '{to_currency}'. Aborting.")
         return amount * from_currency_usd_rate / to_currency_usd_rate
 
-    async def update_exchange_rates_from_coincap(self, session):
+    async def update_exchange_rates_from_price_feeds(self):
```

Comment:
There is a "wait_till_ready" function in `data_feed_base`. Would it be better to wait till the price feed is ready and then copy the exchange rate? Otherwise, there could potentailly be a lot of warning messages